charts/meteor-pipelines/templates/byon-validate-jupyterhub-image.yaml

```diff
@@ -36,3 +36,111 @@ spec:
            echo "    Packages:   Not available"
            echo "{\"severity\":\"error\",\"message\":\"Pip not found\"}" >> $(workspaces.data.path)/messages.ndjson
         fi
+
+    - name: minimal-requirements
+      image: $(params.url)
+      script: |
+        #!/usr/bin/env bash
+        echo "Validating minimal requirements:"
+
+        function ok {
+          echo " ✅"
+        }
+        function skip {
+          echo " ⚠"
+        }
+
+        function fail {
+          echo " ❌"
+          echo "{\"severity\":\"$1\",\"message\":\"$2\"}" >> $(workspaces.data.path)/messages.ndjson
+        }
+
+        # Minimal python version
+        export MINIMAL_PYTHON_VERSION="3.8.0"
+        echo -n "    Python >= $MINIMAL_PYTHON_VERSION"
+        if python -c "from platform import python_version; from packaging.version import parse; from sys import exit; from os import getenv; exit(parse(python_version()) >= parse(getenv('MINIMAL_PYTHON_VERSION')))" 2>/dev/null; then
+          ok
+        else
```

Comment:
is the comparison not matching the if structure maybe? (i.e. should this be `<` instead of `>=`?)

Reply:
Yeah, I've noticed that too. Already resolved with a note above the line explaining the Python+Bash logic. Good observation! :+1: